api/v1beta1/types.go

```diff
@@ -596,21 +596,26 @@ var (
 	InstanceStateUndefined = InstanceState("")
 )
 
-// Bastion represents basic information about the bastion node.
+// Bastion represents basic information about the bastion node. If you enable bastion, the spec has to be specified.
+// +kubebuilder:validation:XValidation:rule="!self.enabled || has(self.spec)",message="you need to specify the spec if bastion is enabled"
 type Bastion struct {
+	// Enabled means that bastion is enabled. Defaults to false.
 	//+optional
```

Comment:
Can we make this required with an explicit default of false?

Reply:
Thanks for this, I wasn't sure how to approach this.